dev/ChangeTracker/IChangeTracker.h

```diff
@@ -10,8 +10,16 @@ namespace winrt::Microsoft::ProjectReunion::implementation
         virtual HRESULT TrackChange(std::function<HRESULT(void)> callBack) = 0;
 
         bool m_ShouldTrackChange{ false };
+        const std::wstring c_UserEvRegLocation{ L"Environment" };
+        const std::wstring c_MachineEvRegLocation{ L"SYSTEM\\CurrentControlSet\\Control\\Session Manager\\Environment" };
         std::wstring m_PackageFullName;
 
+        enum PathOperation
+        {
+            Append,
+            Remove
+        };
+
     private:
         virtual std::wstring KeyName() = 0;
```

Comment:
Sure they can, but it doesn't mean they're virtual chldren of this method.

EnvironmentVariableChangeTracker.h line 19 declares `std::wstring KeyName();`. No virtual keyword and private here should mean no virtual parent method so it's just a method of the subclass. This should fail

```
IChangeTracker ict{ ...get an EnvironmentVariableChangeTracker instance };
...ict.KeyName()...
```

Change to protected or public. Probably the latter. Why shouldn't folks be able to `ichangetracker->KeyName()`?

Also, why is KeyName() non-const? If it changes attributes because it builds the actual value or otherwise caches it, those attributes can be declare `mutable` and then `std::wstring KeyName() const` is fine